libbpf-tools/vfsstat.bpf.c

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: GPL-2.0
+// Copyright (c) 2020 Anton Protopopov
+//
+// Based on vfsstat(8) from BCC by Brendan Gregg
+#include "vmlinux.h"
+#include <bpf/bpf_helpers.h>
+#include <bpf/bpf_tracing.h>
+#include "vfsstat.h"
+
+struct {
+	__uint(type, BPF_MAP_TYPE_ARRAY);
+	__uint(max_entries, S_MAXSTAT);
+	__type(key, int);
+	__type(value, u64);
+} stats SEC(".maps");
```

Comment:
you can use global array instead, given it's just a fixed-sized array. Try:

```
u64 stats[S_MAX_STAT] = {};
```

Then in code just use it as normal array (no bpf_map_lookup_elem/bpf_map_update_elem necesasry). From user-space, you can access it again as a normal array through `obj->bss->stats[i]`.